### DISCUSSION TITLE
Hana cloud DB service provisioning

### DISCUSSION THREAD

**Discussion by I568405** [SAP/terraform-provider-btp#487] [Q&A] [answered]
Hi All,

Is it possible to use terraform-provider-btp to provision Hana Cloud DB ?
I can't seem to find any example on the subject...
Thank you in advance for your help.

**Comment by lechnerc77** [comment, marked as answer]
Hi @Pat-SAP ,

the resource to achieve this is available [btp_subaccount_service_instance](https://registry.terraform.io/providers/SAP/btp/latest/docs/resources/subaccount_service_instance).

As you can see from the examples in the documentation the main "magic" to get an instance up and running are the parameters that must be transferred as a JSON string. We have no dedicated example for HANA DB but the parameters are the same as you would provide for setting up an instance via the BTP CLI. You can check the documentation of SAP HANA Cloud for BTP CLI and transfer that to the parameters for the Terraform provider. You find the documentation here: <https://help.sap.com/docs/hana-cloud/sap-hana-cloud-administration-guide/create-sap-hana-database-instance-using-cli>
